DMCompiler/DM/Expressions/Dereference.cs

```diff
@@ -99,7 +103,8 @@ public Dereference(Location location, DreamPath? path, DMExpression expression,
                     if (callOperation.Safe) {
                         ShortCircuitHandler(proc, endLabel, shortCircuitMode);
                     }
-                    var (argumentsType, argumentStackSize) = callOperation.Parameters.EmitArguments(dmObject, proc);
+                    _valType = proc.ReturnTypes;
```

Comment:
This sets the type to the return types of the current proc, rather than the dereferenced proc